Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+ToolbarDelegate.swift

```diff
@@ -909,46 +910,71 @@ extension BrowserViewController: ToolbarDelegate {
-        handler: UIAction.deferredActionHandler { _ in
-          if let pasteboardContents = UIPasteboard.general.string {
-            self.topToolbar(self.topToolbar, didSubmitText: pasteboardContents)
-          }
-        })
-
-      let pasteAction = UIAction(
-        title: Strings.pasteTitle,
-        image: UIImage(systemName: "doc.on.clipboard"),
-        identifier: .paste,
-        handler: UIAction.deferredActionHandler { _ in
-          if let pasteboardContents = UIPasteboard.general.string {
-            // Enter overlay mode and make the search controller appear.
-            self.topToolbar.enterOverlayMode(pasteboardContents, pasted: true, search: true)
-          }
-        })
-
-      let copyAction = UIAction(
-        title: Strings.copyAddressTitle,
-        image: UIImage(systemName: "doc.on.doc"),
-        handler: UIAction.deferredActionHandler { _ in
-          if let url = self.topToolbar.currentURL {
-            UIPasteboard.general.url = url as URL
+    
+    func createURLBarContextMenuConfiguration(priorityEnabled: Bool) -> UIContextMenuConfiguration {
+      let configuration =  UIContextMenuConfiguration(identifier: nil, previewProvider: nil) { [unowned self] _ in
+        var actionMenu: [UIMenu] = []
+        var pasteMenuChildren: [UIAction] = []
+        
+        let pasteGoAction = UIAction(
+          identifier: .pasteAndGo,
+          handler: UIAction.deferredActionHandler { _ in
+            if let pasteboardContents = UIPasteboard.general.string {
+              self.topToolbar(self.topToolbar, didSubmitText: pasteboardContents)
+            }
+          })
+        
+        let pasteAction = UIAction(
+          identifier: .paste,
+          handler: UIAction.deferredActionHandler { _ in
+            if let pasteboardContents = UIPasteboard.general.string {
+              // Enter overlay mode and make the search controller appear.
+              self.topToolbar.enterOverlayMode(pasteboardContents, pasted: true, search: true)
+            }
+          })
+        
+        pasteMenuChildren = [pasteGoAction, pasteAction]
+        
+        if isUsingBottomBar, !priorityEnabled {
+          pasteMenuChildren.reverse()
+        }
+        
+        let copyAction = UIAction(
+          title: Strings.copyAddressTitle,
+          image: UIImage(systemName: "doc.on.doc"),
+          handler: UIAction.deferredActionHandler { _ in
+            if let url = self.topToolbar.currentURL {
+              UIPasteboard.general.url = url as URL
+            }
+          })
+        
+        let copyMenu = UIMenu(title: "", options: .displayInline, children: [copyAction])
+        
+        if UIPasteboard.general.hasStrings || UIPasteboard.general.hasURLs {
+          let pasteMenu = UIMenu(title: "", options: .displayInline, children: pasteMenuChildren)
+          
+          actionMenu = [pasteMenu, copyMenu]
+          
+          if isUsingBottomBar, !priorityEnabled {
+            actionMenu.reverse()
```

Comment:
This should just check iOS 15 (`if #unavailable(iOS 16.0)`)

Reply:
We also can consider just ignoring iOS 15 entirely and only make this change for iOS 16

Reply:
Fixed at https://github.com/brave/brave-ios/pull/7641/commits/3b4ab2d1f83a9d27b1edb09d04168e9739a7ecbf